core/src/main/scala/org/scalafmt/config/Newlines.scala

```diff
@@ -44,5 +62,18 @@ case class Newlines(
     sometimesBeforeColonInMethodReturnType: Boolean = true,
     penalizeSingleSelectMultiArgList: Boolean = true,
     alwaysBeforeCurlyBraceLambdaParams: Boolean = false,
-    alwaysBeforeTopLevelStatements: Boolean = false
+    alwaysBeforeTopLevelStatements: Boolean = false,
+    afterCurlyLambda: NewlineCurlyLambda = NewlineCurlyLambda.never
 )
+
+sealed abstract class NewlineCurlyLambda
+
+object NewlineCurlyLambda {
+
+  case object preserve extends NewlineCurlyLambda
+  case object always extends NewlineCurlyLambda
+  case object never extends NewlineCurlyLambda
+
+  implicit val newlineCurlyLambdaReader: Reader[NewlineCurlyLambda] =
```

Comment:
I noticed that for every other `Reader` we just define it in the companion object but don't make it implicit, but instead do so in ScalafmtConfig instead. This didn't work for me (due to a macro error I couldn't understand. But i think it wanted the implicit when creating `Newline()` in the config's default param). 

Putting the typeclass instance on the companion object fixes this, and it is proper typeclass useage anyways (avoid orphans).